test/conformance/device_code/CMakeLists.txt

```diff
@@ -81,41 +87,26 @@ macro(add_device_binary SOURCE_FILE)
             continue()
         endif()
 
-        add_custom_command(OUTPUT ${EXE_PATH}
+        add_custom_command(OUTPUT "${BIN_PATH}"
             COMMAND ${UR_DPCXX} -fsycl -fsycl-targets=${TRIPLE} -fsycl-device-code-split=off 
             ${AMD_TARGET_BACKEND} ${AMD_OFFLOAD_ARCH} ${AMD_NOGPULIB}
             ${DPCXX_BUILD_FLAGS_LIST} ${SOURCE_FILE} -o ${EXE_PATH}
 
-            COMMAND ${CMAKE_COMMAND} -E env ${EXTRA_ENV} SYCL_DUMP_IMAGES=true
-            ${EXE_PATH} || exit 0
+            COMMAND ${CMAKE_COMMAND} -E env ${EXTRA_ENV} ${UR_DEVICE_CODE_EXTRACTOR} --stem="${TRIPLE}.bin" ${EXE_PATH}
+
             WORKING_DIRECTORY "${DEVICE_BINARY_DIR}"
             DEPENDS ${SOURCE_FILE}
         )
-        add_custom_target(generate_${KERNEL_NAME}_${TRIPLE} DEPENDS ${EXE_PATH})
+        add_custom_target(generate_${KERNEL_NAME}_${TRIPLE} DEPENDS ${BIN_PATH})
         add_dependencies(generate_device_binaries generate_${KERNEL_NAME}_${TRIPLE})
     endforeach()
     list(APPEND DEVICE_CODE_SOURCES ${SOURCE_FILE})
 endmacro()
 
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/bar.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/device_global.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/fill.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/fill_2d.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/fill_3d.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/fill_usm.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/foo.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/image_copy.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/mean.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/cpy_and_mult.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/cpy_and_mult_usm.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/multiply.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/spec_constant.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/spec_constant_multiple.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/usm_ll.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/saxpy.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/saxpy_usm.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/indexers_usm.cpp)
-add_device_binary(${CMAKE_CURRENT_SOURCE_DIR}/build_failure.cpp)
+file(GLOB SOURCE_FILES "*.cpp")
```

Comment:
Using `GLOB` has caveats, if a new `.cpp` file is added it'll only get picked up when CMake is reconfigured which is not intuitive. This is generally why we've avoided it in favour of using explicit lists of source files.

Reply:
Reverted this.